---
title: "Serratia- and Bacillus-produced Volatiles and their fungistatic properties on Rhizopus stolonifer"
author: Derreck Carter\-House, Josh Chen, Skylar McDonald, Kerry Mauck, Jason Stajich
output:
  pdf_document: 
    keep_tex: yes
    toc: yes
    latex_engine: xelatex
editor_options: 
  chunk_output_type: inline
---


```{r setup, include=FALSE, echo=FALSE}
require("knitr")
opts_knit$set(root.dir = "./")
library(readxl)
library(ggplot2)
library(reshape2)
library(VennDiagram)
library(vegan)
library(ggdendro)
library(png)
library(grid)
library(gridExtra)
library(float)
library(kableExtra)
library(ggpubr)
library(plyr)
library(tiff)
library(cowplot)
options(tinytex.verbose = TRUE)
```

# Colors:
#####  Color Palette by Paletton.com
#####  Palette URL: http://paletton.com/#uid=23E0u0kt+lZlOstrKqzzSiaJidt


*** Primary color:

   shade 0 = #113A73 = rgb( 17, 58,115) = rgba( 17, 58,115,1) = rgb0(0.067,0.227,0.451)
   shade 1 = #385F95 = rgb( 56, 95,149) = rgba( 56, 95,149,1) = rgb0(0.22,0.373,0.584)
   shade 2 = #1D4B8B = rgb( 29, 75,139) = rgba( 29, 75,139,1) = rgb0(0.114,0.294,0.545)
   shade 3 = #072C5F = rgb(  7, 44, 95) = rgba(  7, 44, 95,1) = rgb0(0.027,0.173,0.373)
   shade 4 = #032047 = rgb(  3, 32, 71) = rgba(  3, 32, 71,1) = rgb0(0.012,0.125,0.278)

*** Complement color:

   shade 0 = #AF710B = rgb(175,113, 11) = rgba(175,113, 11,1) = rgb0(0.686,0.443,0.043)
   shade 1 = #E3A948 = rgb(227,169, 72) = rgba(227,169, 72,1) = rgb0(0.89,0.663,0.282)
   shade 2 = #D48F1C = rgb(212,143, 28) = rgba(212,143, 28,1) = rgb0(0.831,0.561,0.11)
   shade 3 = #915A00 = rgb(145, 90,  0) = rgba(145, 90,  0,1) = rgb0(0.569,0.353,0)
   shade 4 = #6B4300 = rgb(107, 67,  0) = rgba(107, 67,  0,1) = rgb0(0.42,0.263,0)


#####  Generated by Paletton.com (c) 2002-2014
```{r colors}
#Divergent:
colors_10=c("#032047","#072C5F","#113A73","#1D4B8B","#385F95","#E3A948","#D48F1C","#AF710B","#915A00","#6B4300")

colors_5blue=c("#032047","#072C5F","#113A73","#1D4B8B","#385F95")
colors_5gold=c("#6B4300","#915A00","#AF710B","#D48F1C","#E3A948")
colors_6div=c("#032047","#113A73","#385F95","#E3A948","#AF710B","#6B4300")
colors_3blue=c("#385F95","#113A73","#032047")
colors_3gold=c("#E3A948", "#AF710B", "#6B4300")
colors_2=c("#E3A948", "#385F95")
```

```{r Sup_fig1, include=FALSE}
library(cowplot)
#Rhizopus data with on S. pro, S. mar, and B. subtilis

# Get Rhizopus data with many strains (48 Hour incubation)
timelap <- read_excel("./BFI_Volatiles.xlsx", sheet = "Serratia Multistrain Analysis", range = "J1:M19")
timelap<-subset(timelap, timelap$Strain=="S. marcescens (ADJS_Red)" |timelap$Strain=="S. proteamaculans")
# Rhizopus name ordering
names.data <- unique(timelap$Strain)
timelap$Strain <- factor(timelap$Strain, levels = names.data)
#boxplot(c ~ Strain, Rhizo, las=2)
names.data

## For loop to obtaim averages and stdv
Strain.unique <- unique(timelap$Strain)
Strain.list <- list()

for (i in 1:length(Strain.unique)){
  # Subset the data frame into each strain
  df.test <- timelap[timelap$Strain %in% Strain.unique[i],]
  # The second element of an apply means: 1 = Rows, 2 = Columns
  # This means we are going to do the mean for all columns in the df
  Strain.list[[i]] <- data.frame(Strain=Strain.unique[i], Time=c("0","24","48"), mean=apply(df.test[,-1],2,mean), sd=apply(df.test[,-1],2,sd))
}

final.data <- as.data.frame(do.call(rbind, Strain.list))
rownames(final.data) <- NULL
#FOR POWERPOINT
#final.data <- subset(final.data, Strain=="E. coli" | Strain=="S. proteamaculans" | Strain=="S. liquefaciens" | Strain=="S. marcescens (ADJS_Red)")
#final.data$Strain <- factor(final.data$Strain,levels = c("E. coli", "S. proteamaculans", "S. liquefaciens", "S. marcescens (ADJS_Red)"))
#final.data$Strain <- factor(final.data$Strain,levels = c("E. coli", "S. proteamaculans", "S. liquefaciens", "S. marcescens (ADJS_Red)"))

Same<-subset(final.data, Time=="0")
oneday<-subset(final.data, Time=="24")
twoday<-subset(final.data, Time=="48")

pd <- position_dodge(1)

timelap.bar.0 <- ggplot(Same, aes(x=Strain, y=mean, fill=Strain)) + geom_bar(stat="identity", position=pd) + geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1, position=pd) + xlab("") + ylab("Percent Growth Inhibition")  +theme(axis.text.x = element_blank(),legend.position = "none")+ scale_fill_manual(values=colors_6div) + ggtitle("24 Hour Fungal Growth with 24-Hour-Old Bacteria")

timelap.bar.24 <- ggplot(oneday, aes(x=Strain, y=mean, fill=Strain)) + geom_bar(stat="identity", position=pd) + geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1, position=pd) + xlab("") + ylab("") + ggtitle("24 Hour Fungal Growth with 48-Hour-Old Bacteria") + theme(axis.text.x = element_blank(),legend.position = "none")+scale_fill_manual(values=colors_6div)+ ylab("Percent Growth Inhibition")

timelap.bar.48 <- ggplot(twoday, aes(x=Strain, y=mean, fill=Strain)) + geom_bar(stat="identity", position=pd) + geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1, position=pd) + xlab("") + ylab("Percent Growth Inhibition") + theme(axis.text.x = element_text(angle=30, hjust=.5, vjust=.5), legend.position = "none")+ scale_fill_manual(values=colors_6div)+ggtitle("24 Hour Fungal Growth with 72-Hour-Old Bacteria")
timelap.bar.48
leg.plot <- ggplot(twoday, aes(x=Strain, y=mean, fill=Strain)) + geom_bar(stat="identity", position=pd) + theme(axis.text.x = element_text(angle = 90, hjust=.5, vjust=1),legend.position = "bottom")+ scale_fill_manual(values=colors_6div)

leg <- get_legend(leg.plot)
plot_grid(timelap.bar.0, timelap.bar.24, timelap.bar.48, nrow=1)

pdf("Rplots/sup_fig1.pdf", width = 6, height=8)
plot_grid(timelap.bar.0, timelap.bar.24, timelap.bar.48,ncol=1, rel_heights = c(.3,.3,.45))
dev.off
#caption: Bacteria grown for 48 hours before \textit{R. stolonifer} were inoculated prevented mycelial growth. n=3
```

```{r include=FALSE}
library(ggplot2)

straw <- read_excel("./BFI_Volatiles.xls", sheet = "Strawberry")

straw.df<- as.data.frame(straw)

straw.sub<-subset(straw.df, straw.df$State =="Control" | straw.df$State == "Smar")
#straw.df$State<- factor(straw.df$State,levels = c("Control", "Smar", "2-Undecanone"))

#my_comparisons <- list( c("Control", "Smar"), c("Control", "2-Undecanone"))

straw.plot <- ggplot(straw.sub, aes(x=State, y=straw.sub$`Number of spores/conidia`, fill=State)) + geom_boxplot() +stat_compare_means(aes(label = ..p.signif..), method = "t.test", ref.group = "Control") + xlab("Condition") + ylab("Spores") +  theme(legend.position = "none") +
  scale_fill_manual(values=colors_2)
straw.plot
#powerpoint
#png("./Rplots/strawplot.png", res=200, height=600, width =800)
#straw.plot
#dev.off()
#dissertation
pdf("./Rplots/strawplot.pdf", height=5, width=5)
straw.plot
dev.off()


```



```{r include=FALSE}
#Pathogens Inhibition
#Fungal data with S. pro, S. mar, and B. subtilis
Donutdata <- read_excel("BFI_Volatiles.xlsx", sheet = "Donut_Plates", range = "A1:F211")
calc_area <- function (d) {
  area <- ((d/2)^2)*pi
  return(area)
}
Donut_area<-NULL
Donut_area$Strain<-Donutdata$Strain
Donut_area$Condition<-Donutdata$Condition
#Donut_area$"0 Hours"<-calc_area(Donutdata$`0 Hours`)
Donut_area$"24 Hours"<-calc_area(Donutdata$`24 Hours`)
Donut_area$"48 Hours"<-calc_area(Donutdata$`48 Hours`)
#Donut_area$"72 Hours"<-calc_area(Donutdata$`72 Hours`)
Donut<-data.frame(Donut_area)
colnames(Donut)<-c("Strain", "Condition", "24 Hours", "48 Hours")

pd=position_dodge(width=1)
Mucor<- subset(Donut, Donut$Strain=="CBS 277.49")
Rhizo<- subset(Donut, Donut$Strain=="NRRL 66455")
Ncra<- subset(Donut, Donut$Strain=="FGSC 4289")
AGB5<- subset(Donut, Donut$Strain=="AGB5")
Conidi<- subset(Donut, Donut$Strain=="ARSEF 448")
Actino<- subset(Donut, Donut$Strain=="NRRL 1706")
Alter<- subset(Donut, Donut$Strain=="Alternaria infectoria")
Donut$"72 Hours"<-calc_area(Donutdata$`72 Hours`)
AF<- subset(Donut, Donut$Strain=="AF 293")

#B. ranarum
AGB5.m <- melt(as.data.frame(AGB5[,2:4]), id.vars = "Condition")
colnames(AGB5.m) <- c("Strain","Time","Growth")
#AGB5.mse <- ddply(AGB5.m, c("Strain","Time") , summarise, N = length(Growth), mean = mean(Growth), sd   = sd(Growth), se   = sd / sqrt(N))
AGB5.m.24<-subset(AGB5.m, AGB5.m$Time=="24 Hours")
AGB5.m.48<-subset(AGB5.m, AGB5.m$Time=="48 Hours")

AGB5.plot.24 <- ggplot(AGB5.m.24, aes(x=Strain, y=Growth, fill=Strain)) +
  geom_boxplot(position=pd)+ 
  theme_minimal() +
  theme(legend.position = "none", axis.text.x = element_blank()) +
  ylab(expression("Area " ~ mm^{2})) +
  xlab("") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B")) + 
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank")

AGB5.plot.48 <- ggplot(AGB5.m.48, aes(x=Strain, y=Growth, fill=Strain))+
  geom_boxplot(position=pd)+ 
  theme_minimal() +
  theme(legend.position = "none", axis.text.x = element_blank())+
  ylab(expression("Area " ~ mm^{2})) +
  xlab("") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B")) + 
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank") 

#A. elegans
Actino.m <- melt(as.data.frame(Actino[,2:4]), id.vars = "Condition")
colnames(Actino.m) <- c("Strain","Time","Growth")
Actino.m.24<-subset(Actino.m, Actino.m$Time=="24 Hours")
Actino.m.48<-subset(Actino.m, Actino.m$Time=="48 Hours")
#Actino.mse <- ddply(Actino.m, c("Strain","Time") , summarise, N = length(Growth), mean = mean(Growth), sd   = sd(Growth), se   = sd / sqrt(N))

Actino.plot.24 <- ggplot(Actino.m.24, aes(x=Strain, y=Growth, fill=Strain)) +
  geom_boxplot(position=pd) +
  theme_minimal() +
  theme(legend.position = "none", axis.text.x = element_blank()) +
  ylab(expression("Area " ~ mm^{2})) +
  xlab("") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B")) +
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank") 

Actino.plot.48 <- ggplot(Actino.m.48, aes(x=Strain, y=Growth, fill=Strain))  +
  geom_boxplot(position=pd)+
  theme_minimal() +
  theme(legend.position = "none", axis.text.x = element_blank()) +
  ylab(expression("Area " ~ mm^{2})) +
  xlab("") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B")) + 
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank") 

#M. circenelloides
Mucor.m <- melt(as.data.frame(Mucor[,2:4]), id.vars = "Condition")
colnames(Mucor.m) <- c("Strain","Time","Growth")
#Mucor.mse <- ddply(Mucor.m, c("Strain","Time") , summarise, N = length(Growth), mean = mean(Growth), sd   = sd(Growth), se   = sd / sqrt(N))
Mucor.m.24<-subset(Mucor.m, Mucor.m$Time=="24 Hours")
Mucor.m.48<-subset(Mucor.m, Mucor.m$Time=="48 Hours")
Mucor.plot.24 <- ggplot(Mucor.m.24, aes(x=Strain, y=Growth, fill=Strain)) +
  geom_boxplot(position=pd)+  
  theme_minimal() +
  theme(legend.position = "none", axis.text.x = element_blank()) + 
  ylab(expression("Area " ~ mm^{2})) +
  xlab("") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B")) +
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank") 

Mucor.plot.48 <- ggplot(Mucor.m.48, aes(x=Strain, y=Growth, fill=Strain))  +
  geom_boxplot(position=pd)+ 
  theme_minimal() +
  theme(legend.position = "none", axis.text.x = element_blank()) +
  ylab(expression("Area " ~ mm^{2})) +
  xlab("") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B")) + 
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank") 

#A. fumigatus
AF.m <- melt(as.data.frame(AF[,2:5]), id.vars = "Condition")
colnames(AF.m) <- c("Strain","Time","Growth")
#AF.mse <- ddply(AF.m, c("Strain","Time") , summarise, N = length(Growth), mean = mean(Growth), sd   = sd(Growth), se   = sd / sqrt(N))
AF.m.24<-subset(AF.m, AF.m$Time=="24 Hours")
AF.m.48<-subset(AF.m, AF.m$Time=="48 Hours")
AF.m.72<-subset(AF.m, AF.m$Time=="72 Hours")

AF.plot.24 <- ggplot(AF.m.24, aes(x=Strain, y=Growth, fill=Strain)) + 
  geom_boxplot(position=pd)+
  theme_minimal() +
  ggtitle("24 Hours") +
  theme(legend.position = "none", plot.title = element_text(hjust = 0.5), axis.text.x = element_blank()) +
  ylab(expression("Area " ~ mm^{2})) +
  xlab("") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B"))+
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank")

AF.plot.48 <- ggplot(AF.m.48, aes(x=Strain, y=Growth, fill=Strain)) +
  geom_boxplot(position=pd)+ 
  theme_minimal() +
  ggtitle("48 Hours") +
  theme(legend.position = "none", plot.title = element_text(hjust = 0.5), axis.text.x = element_blank()) + 
  ylab(expression("Area " ~ mm^{2})) +
  xlab("") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B"))+ 
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank")

AF.plot.72 <- ggplot(AF.m.72, aes(x=Strain, y=Growth, fill=Strain)) +
  geom_boxplot(position=pd)+ 
  theme_minimal() +
  ggtitle("72 Hours") +
  theme(legend.position = "none", plot.title = element_text(hjust = 0.5), axis.text.x = element_blank()) +
  ylab(expression("Area " ~ mm^{2})) +
  xlab("") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B"))+ 
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank")

#R. stolonifer
Rhizo.m <- melt(as.data.frame(Rhizo[,2:4]), id.vars = "Condition")
colnames(Rhizo.m) <- c("Strain","Time","Growth")
#Rhizo.mse <- ddply(Rhizo.m, c("Strain","Time") , summarise, N = length(Growth), mean = mean(Growth), sd   = sd(Growth), se   = sd / sqrt(N))
Rhizo.m.24<-subset(Rhizo.m, Rhizo.m$Time=="24 Hours")
Rhizo.m.48<-subset(Rhizo.m, Rhizo.m$Time=="48 Hours")

Rhizo.plot.24 <- ggplot(Rhizo.m.24, aes(x=Strain, y=Growth, fill=Strain)) + 
  geom_boxplot(position=pd)+
    theme_minimal() +
  theme(legend.position = "none", axis.text.x = element_blank()) +
  ylab(expression("Area " ~ mm^{2})) +
  xlab("") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B"))+
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank")

Rhizo.plot.48 <- ggplot(Rhizo.m.48, aes(x=Strain, y=Growth, fill=Strain)) + 
  geom_boxplot(position=pd)+
    theme_minimal() +
  theme(legend.position = "none", axis.text.x = element_blank()) +
  ylab(expression("Area " ~ mm^{2})) +
  xlab("") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B"))+
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank")

#N. crassa
Ncra.m <- melt(as.data.frame(Ncra[,2:4]), id.vars = "Condition")
colnames(Ncra.m) <- c("Strain","Time","Growth")
#Ncra.mse <- ddply(Ncra.m, c("Strain","Time") , summarise, N = length(Growth), mean = mean(Growth), sd   = sd(Growth), se   = sd / sqrt(N))
Ncra.m.24<-subset(Ncra.m, Ncra.m$Time=="24 Hours")
Ncra.m.48<-subset(Ncra.m, Ncra.m$Time=="48 Hours")

Ncra.plot.24 <- ggplot(Ncra.m.24, aes(x=Strain, y=Growth, fill=Strain)) + 
  geom_boxplot(position=pd)+
  theme_minimal() +
  ggtitle("24 Hours") +
  theme(legend.position = "none", plot.title = element_text(hjust = 0.5), axis.text.x = element_blank()) +
  ylab(expression("Area " ~ mm^{2})) +
  xlab("") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B"))+
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank")

Ncra.plot.48 <- ggplot(Ncra.m.48, aes(x=Strain, y=Growth, fill=Strain)) + 
  geom_boxplot(position=pd)+
  theme_minimal() +
  ggtitle("48 Hours") +
  theme(legend.position = "none", plot.title = element_text(hjust = 0.5), axis.text.x = element_blank()) +
  ylab(expression("Area " ~ mm^{2})) +
  xlab("") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B"))+
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank")


#Legend
leg.plot <- ggplot(Ncra.m.24, aes(x=Strain, y=Growth, fill=Strain)) + 
  geom_boxplot(position=pd)+
    theme_minimal() +
  theme(legend.position = "bottom") +
  ylab("Area (mm^2)")+
  xlab(" ") +
  scale_fill_manual(limits = c("Blank", "Bsub", "Smar", "Spro"), values=c("#D48F1C", "#915A00", "#385F95", "#1D4B8B"))+
  stat_compare_means(label = "p.signif", method = "t.test", ref.group = "Blank")
leg <- get_legend(leg.plot)

fungi_grid_24 <- plot_grid(Ncra.plot.24, Rhizo.plot.24, Mucor.plot.24, AGB5.plot.24, Actino.plot.24, ncol = 1)
fungi_grid_48 <- plot_grid(Ncra.plot.48, Rhizo.plot.48, Mucor.plot.48, AGB5.plot.48, Actino.plot.48, ncol=1)

fungi_grid<-plot_grid(fungi_grid_24,fungi_grid_48, ncol=2)
af<-plot_grid(AF.plot.72, labels = c("F"), ncol =1)
pdf("./Rplots/Donut.pdf", height = 10, width = 6)
plot_grid(fungi_grid, af, leg, ncol = 1, rel_heights = c(1, .2, .1))
dev.off()

#ncrassa 24 inhibition with Spro calc
unique(Donut$Strain)
nc24con<-subset(Donut, Donut$Strain=="FGSC 4289" & Donut$Condition=="Blank")
nc24spro<-subset(Donut, Donut$Strain=="FGSC 4289" & Donut$Condition=="Spro")
mean(nc24spro$`24 Hours`)/mean(nc24con$`24 Hours`)

#Rstol 24 inhibition with Spro calc
unique(Donut$Strain)
rs24con<-subset(Donut, Donut$Strain=="NRRL 66455" & Donut$Condition=="Blank")
rs24spro<-subset(Donut, Donut$Strain=="NRRL 66455" & Donut$Condition=="Spro")
mean(rs24spro$`24 Hours`)/mean(rs24con$`24 Hours`)

#Mucor 24 inhibition with Bsub calc
unique(Donut$Strain)
mc24con<-subset(Donut, Donut$Strain=="CBS 277.49" & Donut$Condition=="Blank")
mc24bsub<-subset(Donut, Donut$Strain=="CBS 277.49" & Donut$Condition=="Bsub")
mean(mc24bsub$`24 Hours`)/mean(mc24con$`24 Hours`)
```

```{r perecent_inhibtion, echo=FALSE}
percent_inhibition <- function(x,y,min) {
  inhib<-100*(1-(x-min)/(y-min))
  return(inhib)
}

loopout<- NULL;
for (i in (unique(Donut$Strain))) {
  f<- subset(Donut, Donut$Strain==i)
  f.m<-melt(as.data.frame(f[,2:6]), id.vars = "Condition")
  colnames(f.m) <- c("Strain","Time","Growth")
  f.mse <- ddply(f.m, c("Strain","Time") , summarise, N = length(Growth), mean = mean(Growth), sd   = sd(Growth), se   = sd / sqrt(N))
  f.mse$Time<-as.character(f.mse$Time)
  initial<-subset(f.mse, Time=="0 Hours" & Strain=="Blank")
  time<-unique(f.mse$Time)
  time<-c(subset(time, time!="0 Hours"))
  for (t in time) {
    strain<-unique(f.mse$Strain)
    strain<-subset(strain, strain!="Blank")
    for (s in strain) {
      x<-subset(f.mse, Time==t & Strain==s)
      y<-subset(f.mse, Time==t & Strain=="Blank")
      results<-percent_inhibition(x$mean,y$mean,initial$mean)
      sd<-100*(sqrt(as.numeric(x$sd)/as.numeric(x$mean))^2)+(sqrt(as.numeric(y$sd)/as.numeric(y$mean))^2)
      df<-data.frame(i,t,s,sd,results)
      loopout<-rbind(loopout,df)
    }
  }
}

colnames(loopout)<-c("Fungi", "Time", "Strain", "Standard Deviation", "Percent Inhibition")
loopout<-subset(loopout, loopout$Fungi!="ARSEF 448")
loopout<-subset(loopout, loopout$Fungi!="Alternaria infectoria")
Bsubdat24<-subset(loopout, loopout$Strain=="Bsub" & loopout$Time=="24 Hours")
Bsubdat48<-subset(loopout, loopout$Strain=="Bsub" & loopout$Time=="48 Hours")
Bsubdat72<-subset(loopout, loopout$Strain=="Bsub" & loopout$Time=="72 Hours")
Smardat24<-subset(loopout, loopout$Strain=="Smar" & loopout$Time=="24 Hours")
Smardat48<-subset(loopout, loopout$Strain=="Smar" & loopout$Time=="48 Hours")
Smardat72<-subset(loopout, loopout$Strain=="Smar" & loopout$Time=="72 Hours")
Sprodat24<-subset(loopout, loopout$Strain=="Spro" & loopout$Time=="24 Hours")
Sprodat48<-subset(loopout, loopout$Strain=="Spro" & loopout$Time=="48 Hours")
Sprodat72<-subset(loopout, loopout$Strain=="Spro" & loopout$Time=="72 Hours")

bsubplot24<-ggplot(Bsubdat24, aes(x=Fungi, y=`Percent Inhibition`, fill=Fungi)) + geom_bar(stat="identity", position = pd, width=0.6) + geom_errorbar(aes(ymin=`Percent Inhibition`-`Standard Deviation`, ymax=`Percent Inhibition`+`Standard Deviation`), width=.2, position=pd) + coord_flip() +scale_y_continuous(limits = c(-30, 120))+xlab("B. subtilis")+ylab("")+ggtitle("24 Hours")+ theme(legend.position = "none", plot.title = element_text(hjust = 0.5))+scale_fill_manual(values = colors_6div)
bsubplot48<-ggplot(Bsubdat48, aes(x=Fungi, y=`Percent Inhibition`, fill=Fungi)) + geom_bar(stat="identity", position = pd, width=0.6) + geom_errorbar(aes(ymin=`Percent Inhibition`-`Standard Deviation`, ymax=`Percent Inhibition`+`Standard Deviation`), width=.2, position=pd)+ coord_flip() +scale_y_continuous(limits = c(-30, 120))+xlab("")+ylab("")+ggtitle("48 Hours")+ theme(legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(hjust = 0.5))+scale_fill_manual(values = colors_6div)
bsubplot72<-ggplot(Bsubdat72, aes(x=Fungi, y=`Percent Inhibition`, fill=Fungi)) + geom_bar(stat="identity", position = pd, width=0.6) + geom_errorbar(aes(ymin=`Percent Inhibition`-`Standard Deviation`, ymax=`Percent Inhibition`+`Standard Deviation`), width=.2, position=pd)+ coord_flip() +scale_y_continuous(limits = c(-30, 120))+xlab("")+ylab("")+ggtitle("72 Hours")+ theme(legend.position = "none", axis.text.y = element_blank(), plot.title = element_text(hjust = 0.5))+scale_fill_manual(values = colors_6div)
bsubplot<-plot_grid(bsubplot24, bsubplot48, bsubplot72, ncol=3, rel_widths = c(4,3,3))

smarplot24<-ggplot(Smardat24, aes(x=Fungi, y=`Percent Inhibition`, fill=Fungi)) + geom_bar(stat="identity", position = pd, width=0.6) + geom_errorbar(aes(ymin=`Percent Inhibition`-`Standard Deviation`, ymax=`Percent Inhibition`+`Standard Deviation`), width=.2, position=pd)+ coord_flip()+scale_y_continuous(limits = c(-30, 120))+ylab("")+xlab("S. marcescens")+ theme(legend.position = "none")+scale_fill_manual(values = colors_6div)
smarplot48<-ggplot(Smardat48, aes(x=Fungi, y=`Percent Inhibition`, fill=Fungi)) + geom_bar(stat="identity", position = pd, width=0.6) + geom_errorbar(aes(ymin=`Percent Inhibition`-`Standard Deviation`, ymax=`Percent Inhibition`+`Standard Deviation`), width=.2, position=pd)+ coord_flip()+scale_y_continuous(limits = c(-30, 120))+xlab("")+ylab("")+ theme(legend.position = "none", axis.text.y = element_blank())+scale_fill_manual(values = colors_6div)
smarplot72<-ggplot(Smardat72, aes(x=Fungi, y=`Percent Inhibition`, fill=Fungi)) + geom_bar(stat="identity", position = pd, width=0.6) + geom_errorbar(aes(ymin=`Percent Inhibition`-`Standard Deviation`, ymax=`Percent Inhibition`+`Standard Deviation`), width=.2, position=pd)+ coord_flip()+scale_y_continuous(limits = c(-30, 120))+xlab("")+ylab("")+ theme(legend.position = "none", axis.text.y = element_blank())+scale_fill_manual(values = colors_6div)
smarplot<-plot_grid(smarplot24, smarplot48, smarplot72, ncol=3, rel_widths = c(4,3,3))

sproplot24<-ggplot(Sprodat24, aes(x=Fungi, y=`Percent Inhibition`, fill=Fungi)) + geom_bar(stat="identity", position = pd, width=0.6)+ geom_errorbar(aes(ymin=`Percent Inhibition`-`Standard Deviation`, ymax=`Percent Inhibition`+`Standard Deviation`), width=.2, position=pd) + coord_flip()+scale_y_continuous(limits = c(-30, 120))+ylab("")+xlab("S. proteamaculans") +theme(legend.position = "none")+scale_fill_manual(values = colors_6div)
sproplot48<-ggplot(Sprodat48, aes(x=Fungi, y=`Percent Inhibition`, fill=Fungi)) + geom_bar(stat="identity", position = pd, width=0.6) + geom_errorbar(aes(ymin=`Percent Inhibition`-`Standard Deviation`, ymax=`Percent Inhibition`+`Standard Deviation`), width=.2, position=pd)+ coord_flip()+scale_y_continuous(limits = c(-30, 120))+xlab("")+ylab("Percent Inhibition")+ theme(legend.position = "none", axis.text.y = element_blank())+scale_fill_manual(values = colors_6div)
sproplot72<-ggplot(Sprodat72, aes(x=Fungi, y=`Percent Inhibition`, fill=Fungi)) + geom_bar(stat="identity", position = pd, width=0.6)+ geom_errorbar(aes(ymin=`Percent Inhibition`-`Standard Deviation`, ymax=`Percent Inhibition`+`Standard Deviation`), width=.2, position=pd) + coord_flip()+scale_y_continuous(limits = c(-30, 120))+xlab("")+ylab("")+ theme(legend.position = "none", axis.text.y = element_blank())+scale_fill_manual(values = colors_6div)
sproplot<-plot_grid(sproplot24, sproplot48, sproplot72, ncol=3, rel_widths = c(4,3,3))

legplot<- ggplot(Sprodat72, aes(x=Fungi, y=`Percent Inhibition`, fill=Fungi)) + geom_bar(stat="identity", position = pd) + theme(legend.position = "bottom")+scale_fill_manual(values = colors_6div)
leg <- get_legend(legplot)
loopout

pdf("./Rplots/Donut_inhibition.pdf", width = 10, height =10)
plot_grid(bsubplot, smarplot, sproplot, leg, rel_heights = c(1,1,1,.2), ncol = 1)
dev.off()

```

```{r include=FALSE}
inhib <- read_excel("BFI_Volatiles.xlsx", sheet = "Donut_percentinhibtion", range = "A44:E68")

stdev <- read_excel("Data_in/Chapter2_data.xlsx", sheet = "Donut_percentinhibtion", range = "A70:E94")

inhib.m <- melt(inhib, id.vars=c("Fungi", "Time"))
stdev.m <- melt(stdev, id.vars=c("Fungi", "Time"))

# 24 Hours
inhib.24 <- subset(inhib.m, Time=="24 Hours")
stdev.24 <- subset(stdev.m, Time=="24 Hours")
Funginames<-unique(inhib.24$Fungi)
myplots.24 <- vector('list', length(Funginames)) 

for (i in Funginames){
    message(i)
    myplots.24[[i]] <- local({
        i <- i
        inhib.24.sub <- subset(inhib.24, Fungi==i)
        stdev.24.sub <- subset(stdev.24, Fungi==i)
        Title<-as.character(i)
        p1 <- ggplot(inhib.24.sub, aes(inhib.24.sub$variable, inhib.24.sub$value, fill=variable)) + geom_bar(stat="identity", position = position_dodge()) + geom_errorbar(aes( ymin=inhib.24.sub$value+stdev.24.sub$value, ymax=inhib.24.sub$value-stdev.24.sub$value), position = position_dodge()) + ylab("Percent Inhibtion") + theme(legend.position = "NULL", axis.text.y = element_blank(), axis.ticks.x = element_blank(), axis.title.x = element_blank()) + scale_fill_manual(values=colors_3blue) + coord_flip() + xlab("") +labs(title=bquote(italic(.(Title)))) 
        print(p1)
    })
}

leg.plot <- ggplot(inhib.24.sub, aes(inhib.24.sub$variable, inhib.24.sub$value, fill=variable)) + geom_bar(stat="identity", position = position_dodge())+ labs(fill="Bacteria")+ scale_fill_manual(values=colors_3blue)
leg <- get_legend(leg.plot)

PercentInhibtion24hours<-plot_grid(myplots.24[["Neurospora crassa"]], myplots.24[["Aspergillus fumigatus"]], myplots.24[["Alternaria infectoria"]], myplots.24[["Rhizopus stolonifer"]], myplots.24[["Mucor circinelloides"]], myplots.24[["Actinomucor elegans"]], myplots.24[["Conidiobolus rhizosporus"]], myplots.24[["Basidiobolus ranarum"]], leg, labels = c("*","","*","*","*","","*(Sp)",""),
  label_size = 10,
  label_x = .91, label_y = .18,
  hjust = 1, vjust = -0.5)
PercentInhibtion24hours

#48 Hours
inhib.48 <- subset(inhib.m, Time=="48 Hours")
stdev.48 <- subset(stdev.m, Time=="48 Hours")
Funginames<-unique(inhib.48$Fungi)
myplots.48 <- vector('list', length(Funginames)) 

for (i in Funginames){
    message(i)
    myplots.48[[i]] <- local({
        i <- i
        inhib.48.sub <- subset(inhib.48, Fungi==i)
        stdev.48.sub <- subset(stdev.48, Fungi==i)
        Title<-as.character(i)
        p1 <- ggplot(inhib.48.sub, aes(inhib.48.sub$variable, inhib.48.sub$value, fill=variable)) + geom_bar(stat="identity", position = position_dodge()) + geom_errorbar(aes( ymin=inhib.48.sub$value+stdev.48.sub$value, ymax=inhib.48.sub$value-stdev.48.sub$value), position = position_dodge()) + ylab("Percent Inhibtion") + theme(legend.position = "NULL", axis.text.y = element_blank(), axis.ticks.x = element_blank(), axis.title.x = element_blank()) + scale_fill_manual(values=colors_3blue) + coord_flip() + xlab("") +labs(title=bquote(italic(.(Title)))) 
        print(p1)
    })
}

leg.plot <- ggplot(inhib.48, aes(inhib.48$variable, inhib.48$value, fill=variable)) + geom_bar(stat="identity", position = position_dodge())+ labs(fill="Bacteria")+ scale_fill_manual(values=colors_3blue)
leg <- get_legend(leg.plot)

PercentInhibtion48hours<-plot_grid(myplots.48[["Neurospora crassa"]], myplots.48[["Aspergillus fumigatus"]], myplots.48[["Alternaria infectoria"]], myplots.48[["Rhizopus stolonifer"]], myplots.48[["Mucor circinelloides"]], myplots.48[["Actinomucor elegans"]], myplots.48[["Conidiobolus rhizosporus"]], myplots.48[["Basidiobolus ranarum"]], leg, labels = c("*","","*(Sp)","*(Sp/Sm)","*(Sp/Sm)","","*(Sp)","*"),
  label_size = 10,
  label_x = .91, label_y = .2,
  hjust = 1, vjust = -0.5)


#72 Hours
inhib.72 <- subset(inhib.m, Time=="72 Hours")
stdev.72 <- subset(stdev.m, Time=="72 Hours")
Funginames<-unique(inhib.72$Fungi)
myplots.72 <- vector('list', length(Funginames)) 

for (i in Funginames){
    message(i)
    myplots.72[[i]] <- local({
        i <- i
        inhib.72.sub <- subset(inhib.72, Fungi==i)
        stdev.72.sub <- subset(stdev.72, Fungi==i)
        Title<-as.character(i)
        p1 <- ggplot(inhib.72.sub, aes(inhib.72.sub$variable, inhib.72.sub$value, fill=variable)) + geom_bar(stat="identity", position = position_dodge()) + geom_errorbar(aes( ymin=inhib.72.sub$value+stdev.72.sub$value, ymax=inhib.72.sub$value-stdev.72.sub$value), position = position_dodge()) + ylab("Percent Inhibtion") + theme(legend.position = "NULL", axis.text.y = element_blank(), axis.ticks.x = element_blank(), axis.title.x = element_blank()) + scale_fill_manual(values=colors_3blue) + coord_flip() + xlab("") +labs(title=bquote(italic(.(Title)))) 
        print(p1)
    })
}

leg.plot <- ggplot(inhib.72, aes(inhib.72$variable, inhib.72$value, fill=variable)) + geom_bar(stat="identity", position = position_dodge())+ labs(fill="Bacteria")+ scale_fill_manual(values=colors_3blue)
leg <- get_legend(leg.plot)

PercentInhibtion72hours<-plot_grid(myplots.72[["Neurospora crassa"]], myplots.72[["Aspergillus fumigatus"]], myplots.72[["Alternaria infectoria"]], myplots.72[["Rhizopus stolonifer"]], myplots.72[["Mucor circinelloides"]], myplots.72[["Actinomucor elegans"]], myplots.72[["Conidiobolus rhizosporus"]], myplots.72[["Basidiobolus ranarum"]], leg, labels = c("*(Sp)","*","","","*(Sp)","","*(Sp)","*(Sm)"),
  label_size = 10,
  label_x = .7, label_y = .2,
  hjust = -0.5, vjust = -0.5)

  
pdf("./R-plots/PercentInhibtion24hours.pdf", width=8, height = 8)
PercentInhibtion24hours
dev.off()

pdf("./R-plots/PercentInhibtion48hours.pdf", width=8, height = 8)
PercentInhibtion48hours
dev.off()

pdf("./R-plots/PercentInhibtion72hours.pdf", width=8, height = 8)
PercentInhibtion72hours
dev.off()


```

```{r include=FALSE}

Samples <- read_excel("BFI_Volatiles.xlsx", sheet = "Total_production")

Samples.df <- as.data.frame(Samples)

rownames(Samples.df) <- Samples$`Sample`

vol_total <- ggplot(Samples.df, aes(x=Samples.df$`Sample`, y=Samples.df$`Detected Volatiles (ng)`, fill=Samples.df$`Detected Volatiles (ng)`)) + geom_bar(stat="identity") + guides(fill=FALSE) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + ylab("ng of Volatiles") + xlab("Sample") + scale_fill_gradient2(low="#113A73", mid="#AF710B", high="#E3A948", midpoint = 1000) + geom_hline(yintercept=200, linetype="dotted")
vol_total

```

```{r totes, echo=FALSE, warning=FALSE, out.width = "90%", fig.pos="H",fig.cap="Effect of the most abundance VOCs produced by bacteria on Rhizopus growth after 24 hours"}
pdf("./Rplots/totalGCMS.pdf", heigh=5, width=8)
vol_total
dev.off()
```

```{r boxplot_vol, echo=FALSE, out.width = "90%", fig.pos = "H", fig.cap="Table of differentially produced volatiles"}
library(reshape2)
library(readxl)
gcdata <- read_excel("BFI_Volatiles.xlsx", 
    sheet = "GCdata")
gcdata.85<-data.frame(subset(gcdata, gcdata$RSI>=850))
gcdata.85.bsub<-subset(gcdata.85, gcdata.85$Bacteria=="Bsub")
gcdata.85.smar<-subset(gcdata.85, gcdata.85$Bacteria=="Smar")
gcdata.85.spro<-subset(gcdata.85, gcdata.85$Bacteria=="Spro")

nosignletons.bsub<-gcdata.85.bsub[gcdata.85.bsub$Putative.compound %in% gcdata.85.bsub$Putative.compound[duplicated(gcdata.85.bsub$Putative.compound)],]
nrow(gcdata.85.bsub)
nrow(nosignletons.bsub)

nosignletons.smar<-gcdata.85.smar[gcdata.85.smar$Putative.compound %in% gcdata.85.smar$Putative.compound[duplicated(gcdata.85.smar$Putative.compound)],]
nrow(gcdata.85.smar)
nrow(nosignletons.smar)

nosignletons.spro<-gcdata.85.spro[gcdata.85.spro$Putative.compound %in% gcdata.85.spro$Putative.compound[duplicated(gcdata.85.spro$Putative.compound)],]
nrow(gcdata.85.spro)
nrow(nosignletons.spro)


compds.plot.bsub<-ggplot(nosignletons.bsub, aes(x=Putative.compound, y=Amount..ng.per.total.collection.time.)) + 
  geom_boxplot(position = position_dodge()) + 
  theme_minimal() +
  ylab("Nanograms detected") + 
  xlab("") + 
  ggtitle(expression(paste(italic("B. subtils")))) +
  coord_flip() 

compds.plot.smar<-ggplot(nosignletons.smar, aes(x=Putative.compound, y=Amount..ng.per.total.collection.time.)) + 
  geom_boxplot(position = position_dodge()) + 
  theme_minimal() +
  ylab("Nanograms detected") + 
  xlab("") + 
  ggtitle(expression(paste(italic("S. marcescens")))) +
  coord_flip()

compds.plot.spro<-ggplot(nosignletons.spro, aes(x=Putative.compound, y=Amount..ng.per.total.collection.time.)) + 
  geom_boxplot(position = position_dodge()) + 
  theme_minimal() +
  ylab("Nanograms detected") + 
  xlab("") + 
  ggtitle(expression(paste(italic("S. proteamaculans")))) +
  coord_flip()

pdf("./Rplots/abund_compds.pdf", height = 8, width =11)
plot_grid(vol_total,compds.plot.bsub, compds.plot.smar, compds.plot.spro, ncol=2, labels = c("A", "B", "C", "D"))
dev.off()
```

```{r upsetr}
library(UpSetR)
library(reshape2)
gcdata <- read_excel("BFI_Volatiles.xlsx", 
    sheet = "GCdata")
gcdata.85<-subset(gcdata, gcdata$RSI>=850)
upset.data<-gcdata.85[,c(2,4,10)]
upset.data$`Amount (ng per total collection time)`[upset.data$`Amount (ng per total collection time)` > 0] <- 1
upset.table<-table(upset.data)
upset.df<-data.frame(upset.table)
upset.df<-upset.df[,c(1,2,4)]
upset.c<-dcast(upset.df, Bacteria~`Putative.compound`)
upset.t<-t(upset.c)
upset.df<-as.data.frame(upset.t)
upset.df2<-upset.df[2:nrow(upset.df),]

upset.df2[2:4]<-upset.df2[1:3]
colnames(upset.df2)<-c("Compound", "Bsub", "Smar", "Spro")

upset.df2$Bsub<-as.integer(upset.df2$Bsub)
upset.df2$Smar<-as.integer(upset.df2$Smar)
upset.df2$Spro<-as.integer(upset.df2$Spro)

upset.df2[upset.df2 < 4] <- 0
upset.df2[upset.df2 > 0] <- 1
upset.df2[1]<-rownames(upset.df2)
rownames(upset.df2)<-NULL

upset(upset.df2, order.by = "freq")

pdf("./Rplots/upsetr.pdf", height = 4, width = 4)
upset(upset.df2)
dev.off()

```


```{r venndia, echo=FALSE,fig.pos="H", out.width = "90%",fig.cap="Over lap of related volatiles produced by three species of bacteria"}



grid.newpage()
  Venn <- draw.triple.venn(area1 = 14, area2 = 16, area3 = 11, n12 = 3, n23 = 9, n13 = 2, n123 = 2, category = c("Bsub", "Smar", "Spro"), lty = rep("solid", 3), fill = c("#D48F1C", "#385F95", "#072C5F"), scaled = TRUE, euler.d = TRUE)



pdf("./R-plots/Venn.pdf", height = 4, width = 4)

grid.newpage()
  Venn <- draw.triple.venn(area1 = 14, area2 = 16, area3 = 11, n12 = 3, n23 = 9, n13 = 2, n123 = 2, category = c("Bsub", "Smar", "Spro"), lty = rep("solid", 3), fill = c("#D48F1C", "#385F95", "#072C5F"), scaled = TRUE, euler.d = TRUE)
  
dev.off()


png("./R-plots/Venn.png", height = 400, width = 500, res=150)

grid.newpage()
  Venn <- draw.triple.venn(area1 = 14, area2 = 16, area3 = 11, n12 = 3, n23 = 9, n13 = 2, n123 = 2, category = c("Bsub", "Smar", "Spro"), lty = rep("solid", 3), fill =c("#56B4E9", "#E69F00", "#999999"))
  
dev.off()

```


```{r include=FALSE}
#NMDS Plot
library(vegan)
volatiles <- read_excel("BFI_Volatiles.xlsx", sheet = "NMDS_1")
volatiles_names <- read_excel("BFI_Volatiles.xlsx", sheet = "NMDS_2")
volatiles
rownames(volatiles) <- volatiles_names$Sample

volatiles.matrix=as.matrix(volatiles)

vol.rel <- decostand(volatiles.matrix, method = "total")
vol.rel_distmat <- vegdist(vol.rel, method = "bray")
vol.rel_distmat <- as.matrix(vol.rel_distmat, labels = T)
write.csv(vol.rel_distmat, "volatiles_relative_distance_matrix_nmds.csv")

vol.rel_NMS <- metaMDS(vol.rel_distmat, distance = "bray", k = 2, maxit = 999, trymax = 500, wascores = TRUE)

goodness(vol.rel_NMS)
stressplot(vol.rel_NMS)
plot(vol.rel_NMS)

``` 

```{r nmds, echo=FALSE, out.width = "90%", fig.pos="H",fig.cap="NMDS showing differences between Bacillus and Serratia"}

treat=c(rep("Bsub",6),rep("Smar",6),rep("Spro",8))
ordiplot(vol.rel_NMS,type="points")
ordihull(vol.rel_NMS,groups=treat,draw="polygon",col=c("#D48F1C", "skyblue", "#072C5F"),label=F, scaling = 3, main = expression(air == 2), cex = 2)
legend(x="bottomleft", legend=unique(treat), col=c("#D48F1C", "skyblue", "#072C5F"), pch=16)

pdf("./Rplots/NMDS.pdf", height=5, width=4)
treat=c(rep("Bsub",6),rep("Smar",6),rep("Spro",8))
ordiplot(vol.rel_NMS,type="points")
ordihull(vol.rel_NMS,groups=treat,draw="polygon",col=c("#D48F1C", "skyblue", "#072C5F"),label=F, scaling = 3, main = expression(air == 2), cex = 2)
legend(x="bottomleft", legend=unique(treat), col=c("#D48F1C", "skyblue", "#072C5F"), pch=16)
dev.off()

```

```{r metabolanalyst_nooutliers, echo=FALSE, include=FALSE}
library(readxl)
gcdata <- read_excel("BFI_Volatiles.xlsx", 
    sheet = "GCdata")
gcdata.85<-subset(gcdata, gcdata$RSI>=850)
metabo.data<-dcast(gcdata.85, Sample~`Putative compound`, fun.aggregate = sum)
metabo.data$Bacteria<- substr(metabo.data$Sample,1,nchar(metabo.data$Sample)-2)
#write.csv(metabo.data, "Metaboanalyst_in.csv")
metanr_packages <- function(){
  
metr_pkgs <- c("Rserve", "ellipse", "scatterplot3d", "Cairo", "randomForest", "caTools", "e1071", "som", "impute", "pcaMethods", "RJSONIO", "ROCR", "globaltest", "GlobalAncova", "Rgraphviz", "preprocessCore", "genefilter", "pheatmap", "SSPA", "sva", "Rcpp", "pROC", "data.table", "limma", "car", "fitdistrplus", "lars", "Hmisc", "magrittr", "methods", "xtable", "pls", "caret", "lattice", "igraph", "gplots", "KEGGgraph", "reshape", "RColorBrewer", "tibble", "siggenes", "plotly", "xcms", "CAMERA", "fgsea", "MSnbase", "BiocParallel", "metap", "reshape2", "scales")
  
  list_installed <- installed.packages()
  
  new_pkgs <- subset(metr_pkgs, !(metr_pkgs %in% list_installed[, "Package"]))
  
  if(length(new_pkgs)!=0){
    
    if (!requireNamespace("BiocManager", quietly = TRUE))
      install.packages("BiocManager")
    BiocManager::install(new_pkgs, version = "3.8")
    print(c(new_pkgs, " packages added..."))
  }
  
  if((length(new_pkgs)<1)){
    print("No new packages added...")
  }
}

metanr_packages()

if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
#BiocManager::install(c("plotly", "texi2dvi", "impute", "pcaMethods", "globaltest", "GlobalAncova", "Rgraphviz", "preprocessCore", "genefilter", "SSPA", "sva", "limma", "KEGGgraph", "siggenes", "xcms", "CAMERA", "fgsea", "MSnbase", "BiocParallel"))
#devtools::install_cran("latexpdf")
#install.packages("devtools")
#library(devtools)

#devtools::install_github("xia-lab/MetaboAnalystR", build = TRUE, build_opts = c("--no-resave-data", "--no-manual"))

library(MetaboAnalystR)

mSet<-InitDataObjects("conc", "stat", FALSE)
mSet<-Read.TextData(mSet, "./Metaboanalyst_in.csv", "rowu", "disc");
mSet<-SanityCheckData(mSet)
mSet<-ReplaceMin(mSet);
mSet<-PreparePrenormData(mSet)
mSet<-Normalization(mSet, "NULL", "NULL", "AutoNorm", ratio=FALSE, ratioNum=20)
mSet<-PlotNormSummary(mSet, "norm_2_", "png", 600, width=NA)
mSet<-PlotSampleNormSummary(mSet, "snorm_2_", "png", 600, width=NA)
mSet<-ANOVA.Anal(mSet, F, 0.05, "fisher", FALSE)
mSet<-PlotANOVA(mSet, "aov_2_", "png", 600, width=NA)
mSet<-ANOVA.Anal(mSet, F, 0.05, "tukey", FALSE)
mSet<-PlotANOVA(mSet, "aov_3_", "png", 600, width=NA)
und<-PlotCmpdView(mSet, "2-Undecanone", "pdf", 150, width=NA)
dmts<-PlotCmpdView(mSet, "Dimethyl trisulfide", "png", 150, width=NA)
ani<-PlotCmpdView(mSet, "Anisole", "png", 150, width=NA)
mSet<-PCA.Anal(mSet)
mSet<-PlotPCAPairSummary(mSet, "pca_pair_2_", "png", 600, width=NA, 5)
mSet<-PlotPCAScree(mSet, "pca_scree_2_", "png", 600, width=NA, 5)
mSet<-PlotPCA2DScore(mSet, "pca_score2d_2_", "png", 600, width=NA, 1,2,0.95,0,0)
mSet<-PlotPCALoading(mSet, "pca_loading_2_", "png", 600, width=NA, 1,2);
mSet<-PlotPCABiplot(mSet, "pca_biplot_2_", "png", 600, width=NA, 1,2)
mSet<-PlotPCA3DScoreImg(mSet, "pca_score3d_2_", "png", 600, width=NA, 1,2,3, 40)
#mSet<-PlotPCA3DLoading(mSet, "pca_loading3d_2_", "json", 1,2,3)
mSet<-PlotHeatMap(mSet, "heatmap_1_", "png", 72, width=NA, "norm", "row", "euclidean", "ward.D","bwm", "overview", T, T, NA, T, F)
#mSet<-PlotSubHeatMap(mSet, "heatmap_3_", "png", 600, width=NA, "norm", "row", "euclidean", "ward.D","bwm", "tanova", 58, "overview", T, T, T, F)
#mSet<-PlotSubHeatMap(mSet, "heatmap_3_", "png", 600, width=NA, "norm", "row", "correlation", "ward.D","bwm", "tanova", 58, "overview", T, T, T, F)

mSet<-PlotSubHeatMap(mSet, "heatmap_3_", "png", 600, width=NA, "norm", "row", "correlation", "average","bwy", "tanova", 63, "overview", T, T, T, F)
mSet<-PlotHeatMap(mSet, "heatmap_2_", "png", 600, width=NA, "norm", "row", "correlation", "average","bwm", "overview", T, T, NA, T, F)
mSet<-RF.Anal(mSet, 500,7,1)
mSet<-PlotRF.Classify(mSet, "rf_cls_2_", "png", 72, width=NA)
mSet<-PlotRF.VIP(mSet, "rf_imp_2_", "png", 72, width=NA)
mSet<-PlotRF.Outlier(mSet, "rf_outlier_2_", "png", 72, width=NA)
mSet<-RF.Anal(mSet, 5000,13,1)
mSet<-PlotRF.Classify(mSet, "rf_cls_3_", "png", 600, width=NA)
mSet<-PlotRF.VIP(mSet, "rf_imp_3_", "png", 72, width=NA)
mSet<-PlotRF.Outlier(mSet, "rf_outlier_3_", "png", 600, width=NA)
mSet<-PlotHCTree(mSet, "tree_2_", "png", 600, width=NA, "spearman", "average")

img1 <-  rasterGrob(as.raster(readPNG("./Dimethyl trisulfide_dpi72.png")), interpolate = FALSE)
img2 <-  rasterGrob(as.raster(readPNG("./2-Undecanone_dpi72.png")), interpolate = FALSE)
img3 <-  rasterGrob(as.raster(readPNG("./Anisole_dpi72.png")), interpolate = FALSE)

png("./Rplots/difexprvol.png", width = 960, res = 300)
grid.arrange(img1, img2, img3, nrow = 1)
dev.off
```


```{r pca, echo=FALSE, out.width = "90%", fig.pos="H",fig.cap="PCA plot showing variation amount samples of Bsub, Smar, and Spro"}

knitr::include_graphics("pca_score3d_2_dpi600.png")
```

```{r meta_heat, echo=FALSE, out.width = "90%", fig.pos="H", fig.cap="Shows autoscaled volatiles and the production relative to each sample"}
knitr::include_graphics("./heatmap_3_dpi600.png")
```

```{r meta_heat_ave, echo=FALSE, out.width = "90%", fig.pos="H", fig.cap="Average of volatiles produced by each species of bacteria"}
knitr::include_graphics("./R-plots/heatmap_7_dpi72.png")
```

```{r asdf, echo=FALSE, out.width = "90%", fig.pos="H", fig.cap="ANOVA results showing 95% confidence or higher 11 compounds differentially produced by Bacteria"}
knitr::include_graphics("./R-plots/aov_2_dpi600.png")
```

```{r anovatable, echo=FALSE, out.width = "90%", fig.pos = "H", fig.cap="Table of differentially produced volatiles"}

anov_res <- read_excel("BFI_Volatiles.xlsx", sheet = "ANOVA")
anov_res <- as.data.frame(anov_res)

library(kableExtra)
#webshot::install_phantomjs()
table_vol <- anov_res %>%
  kable() %>%
  kable_styling(bootstrap_options = c("striped", "hover"))%>%
  save_kable("./R-plots/ANOVA-table.pdf")


```


```{r anova_compounds, echo=FALSE, out.width = "100%", fig.pos="H", fig.cap="Relationship of differentially produced volatiles between each bacteria species"}
grid.arrange(img1, img2, img3, nrow = 3)

```


```{r include=FALSE}
# Pure volatile inhibition
#Get pure volatile assay data
library(ggpubr)
abund <- read_excel("BFI_Volatiles.xlsx", sheet = "Abundant_VOC_RS_24")
abund$Compound <- factor(abund$Compound, levels = c("Dimethyl trisulfide","Anisole", "2-Undecanone", "2,5-Dimethylpyrazine",  "Lepidine", "5-Methyl-2-furylmethanol","Tropone","Unexposed"))

calc_area <- function (d) {
  area <- ((d/2)^2)*pi
  return(area)
}
abund$area<-calc_area(abund$Diameter)

abund.plot<-ggplot(abund, aes(x=Compound, y=area, fill=Compound)) +
  geom_boxplot() + 
  ylab(expression(paste("Area of ", italic("R. stolonifer"), " mycelium"  ~ cm^{2}))) + 
  xlab("10 mg of Compound") +  
  guides(fill=FALSE)+scale_fill_manual(values=colors_10)+
  annotate("text", y=7, x=c("Dimethyl trisulfide","Anisole", "2-Undecanone") ,label = "****") +
  annotate("text", y=7, x=c("2,5-Dimethylpyrazine",  "Lepidine") ,label = "***") +
  annotate("text", y=7, x=c("5-Methyl-2-furylmethanol") ,label = "**") +
  coord_flip()

x=subset(abund$area, abund$Compound=="Unexposed")
calc_p <- function (x,y) {
  p <- t.test(x,y)
  return(p$p.value)
}
comps_list<-unique(abund$Compound)
for (c in 1:length(comps_list)){
  y=subset(abund$area, abund$Compound==comps_list[c])
  print(comps_list[c])
  print(calc_p(x,y))
}
pdf("Rplots/VolandStraw.pdf", height = 4, width = 8)
plot_grid(abund.plot, straw.plot, labels = c("A", "B"), ncol=2, rel_widths = c(1, .6))
dev.off()

```

```{r abundance, echo=FALSE, out.width = "90%", fig.pos="H",fig.cap="Growth measurements after 24 hours of exposure to 10mg of each compound"}
abund.plot
pdf("./Rplots/vol_mycelia.pdf", height = 4, width =4.5)
abund.plot 
dev.off()
```


```{r microscopy, echo=FALSE, out.width="90%", fig.cap="Panel of Microscope images"}
knitr::opts_chunk$set(dpi=300,fig.width=7)
micro_con <- rasterGrob(as.raster(readPNG("images/40x_con_6.png")), interpolate = FALSE)
micro_dmpz <-  rasterGrob(as.raster(readPNG("images/40x_dmpz_7.png")), interpolate = FALSE)
micro_und <-  rasterGrob(as.raster(readPNG("images/40x_und_5.png")), interpolate = FALSE)
micro_an <-  rasterGrob(as.raster(readPNG("images/40x_An_1.png")), interpolate = FALSE)
micro_dmts <-  rasterGrob(as.raster(readPNG("images/40x_DMTS_7.png")), interpolate = FALSE)

library(cowplot)

title = cowplot::ggdraw() + cowplot::draw_label("40X "~italic(Rhizopus)~ " Hyphae Exposed to Volatiles", size = 20)
top_row = cowplot::plot_grid(micro_con, micro_dmpz, micro_und, labels = c('A', 'B', 'C'), ncol=3)
bottom_row = cowplot::plot_grid(micro_an, micro_dmts, labels = c('D', 'E'), ncol=2)

plot_grid(top_row, bottom_row, ncol=2)

png("R-plots/Micro.png", height = 2400, width =2400, res=600)
plot_grid(top_row, bottom_row, ncol=1)
dev.off()


micro_nc_con <- rasterGrob(as.raster(readPNG("./images/NC_Control.png")), interpolate = FALSE)
micro_nc_und <-  rasterGrob(as.raster(readPNG("./images/NC_und.png")), interpolate = FALSE)
plot_grid(micro_nc_con, micro_nc_und, labels = c('A', 'B'), ncol=1)

png("./R-plots/NC_40x.png", height = 2400, width =2400, res=600)
plot_grid(micro_nc_con, micro_nc_und, labels = c('A', 'B'), ncol=1)
dev.off()
```

```{r include=FALSE}
library(readxl)
library(ggplot2)
straw <- read_excel("BFI_Volatiles.xlsx", sheet = "Strawberry")

straw.df<- as.data.frame(straw)

straw.sub<-subset(straw.df, straw.df$State =="Con" | straw.df$State == "Spro")
straw.sub

straw.plot2 <- ggplot(straw.sub, aes(x=State, y=`Number of spores/conidia`, fill=State)) + geom_boxplot() + stat_compare_means(method = "t.test", label.x = c("Spro"), label.y = 200) + xlab("Condition") + theme(legend.position = "none") + ggtitle("2-Undecanone Reduces Rstol Spore Production On Strawberry")
straw.plot2
```

```{r strawberry, echo=FALSE, out.width="90%", fig.cap="Spores recovered from strawberries innoculated with Rstol in the presence of S. marcescens"}

png("R-plots/StrawSpore2.png")
straw.plot2
dev.off()
```


```{r genus_volatiles, include=FALSE}

genus_gc <- read_excel("BFI_Volatiles.xlsx", sheet = "Comparative_Genus")
genus_gc

# Create dendro
rownames(genus_gc) <- genus_gc$Strain

gc.dendro <- as.dendrogram(hclust(d = dist(x = genus_gc)))
dendro.plot <- ggdendrogram(data = gc.dendro, rotate = TRUE)

# Preview the plot
dendro.plot

#make heatmap

gc.long <- melt(genus_gc, id = c("Strain", "Genus"))

#ggplot(genus_gc.m, aes(Samples, variable, fill=value)) + geom_tile()


heatmap.plot <- ggplot(data = gc.long, aes(x = Strain, y = variable)) + geom_point(alpha=.7, aes(color=Genus, size=as.character(gc.long$value)), shape=20) + ylab("Compound") + xlab("Bacteria strains") + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + labs(fill = "Probability of Presence")+ scale_size_manual(values=c(0,5)) + guides(size = FALSE)

heatmap.plot
pdf("./R-plots/Genus_volatiles.pdf", width = 8, height =10)
heatmap.plot
dev.off()

#Order Branches
branch.order <- order.dendrogram(gc.dendro)
gc.long$Samples <- factor(x = gc.long$Strain,
                                levels = unique(gc.long$Strain[branch.order]), ordered = TRUE)


#make sorted heatmap
gc.long
heatmap.plot.sorted <- ggplot(data = gc.long, aes(x = Strain, y = variable)) +
  geom_tile(aes(fill = value)) +
  scale_fill_gradient(low = "grey", high = "blue", na.value = "black") +  ylab("Compound") +
  xlab("List of Strains") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(fill = "Probability of Presence")

pdf("Genus_volatiles.pdf", height = 16, width = 16)
heatmap.plot.sorted
dev.off()
```

# Discussion:

The biggest impact on the field. A reexamination of volatiles produced by Serratia marcescens reveals several antifungal compounds. An issue with 2-undecanone is that it has a very short halflife of 1.2 days [76].
The context of impact within the field - Many studies are beginning to incorporate microbial communities into their work, the volatile profiles of these communities are at times neglected but can clearly play a major role. 
Support from other work includes microbial community assembly work is being done, this adds to the body of work showing the importance of fungal bacteria interplay [1].
Limitations of the work are that we are limited to the bacterial profile being created while bacteria are on LB media, hardly a realistic environmental conditions. It would be interesting to see the volatiles produced in various soil types and with various plant partners to see if there is a core volatome. 

# Conclusion:

It appears the key to resisting the bacteria compounds is to metabolize or neutralize the volatiles. More work needs to be done in other systems, especially outside of a petri dish. Spores/Conidia can sense the presence of volatiles and wait until favorable conditions, this would be an interesting follow-up experiment to learn about environmental cues that stop fungi from germinating from conidia. 
Looking into additional compounds to find more fungal responses or if there is a conserved fungal response. Anisole would be an interesting compound to follow up on, only found in Serratia marcescens (ADJS-Red). Not in Bacillus. 
Searching for new antifungals is important. Looking to the future, volatile profiles with multiple antifungal components may be important for the war against fungi. 



# References


1. 	Zhang Y, Kastman EK, Guasto JS, Wolfe BE. Fungal networks shape dynamics of bacterial dispersal and community assembly in cheese rind microbiomes. Nat Commun. 2018;9: 336.
2. 	Fleming A. On the Antibacterial Action of Cultures of a Penicillium, with Special Reference to their Use in the Isolation of B. influenz. Br J Exp Pathol. 1929;10: 226.
3. 	Meiser A, Otte J, Schmitt I, Grande FD. Sequencing genomes from mixed DNA samples - evaluating the metagenome skimming approach in lichenized fungi. Sci Rep. 2017;7: 14881.
4. 	Bahram M, Hildebrand F, Forslund SK, Anderson JL, Soudzilovskaia NA, Bodegom PM, et al. Structure and function of the global topsoil microbiome. Nature. 2018;560: 233237.
5. 	Gonzalez E, Pitre FE, Pag AP, Marleau J, Guidi Nissim W, St-Arnaud M, et al. Trees, fungi and bacteria: tripartite metatranscriptomics of a root microbiome responding to soil contamination. Microbiome. 2018;6: 53.
6. 	Schroeckh V, Scherlach K, Ntzmann H-W, Shelest E, Schmidt-Heck W, Schuemann J, et al. Intimate bacterial-fungal interaction triggers biosynthesis of archetypal polyketides in Aspergillus nidulans. Proc Natl Acad Sci U S A. 2009;106: 1455814563.
7. 	Partida-Martinez LP, Hertweck C. Pathogenic fungus harbours endosymbiotic bacteria for toxin production. Nature. 2005;437: 884888.
8. 	Uehling J, Gryganskyi A, Hameed K, Tschaplinski T, Misztal PK, Wu S, et al. Comparative genomics of Mortierella elongata and its bacterial endosymbiont Mycoavidus cysteinexigens. Environ Microbiol. 2017; doi:10.1111/1462-2920.13669
9. 	Li Z, Yao Q, Dearth SP, Entler MR, Castro Gonzalez HF, Uehling JK, et al. Integrated proteomics and metabolomics suggests symbiotic metabolism and multimodal regulation in a fungal-endobacterial system. Environ Microbiol. 2017;19: 10411053.
10. Desir A, Hao Z, Liber JA, Benucci GMN, Lowry D, Roberson R, et al. Mycoplasma-related endobacteria within Mortierellomycotina fungi: diversity, distribution and functional insights into their lifestyle. ISME J. 2018;12: 17431757.
11. Hoffman MT, Arnold AE. Diverse bacteria inhabit living hyphae of phylogenetically diverse fungal endophytes. Appl Environ Microbiol. 2010;76: 40634075.
12. Partida-Martinez LP, Monajembashi S, Greulich K-O, Hertweck C. Endosymbiont-dependent host reproduction maintains bacterial-fungal mutualism. Curr Biol. 2007;17: 773777.
13. Shaffer JP, URen JM, Gallery RE, Baltrus DA, Arnold AE. An Endohyphal Bacterium (Chitinophaga, Bacteroidetes) Alters Carbon Source Use by Fusarium keratoplasticum (F. solani Species Complex, Nectriaceae). Front Microbiol. 2017;8: 350.
14. Pawlowska TE, Gaspar ML, Lastovetsky OA, Mondo SJ, Real-Ramirez I, Shakya E, et al. Biology of Fungi and Their Bacterial Endosymbionts. Annu Rev Phytopathol. 2018;56: 289309.
15. Deveau A, Bonito G, Uehling J, Paoletti M, Becker M, Bindschedler S, et al. Bacterial - Fungal Interactions: ecology, mechanisms and challenges. FEMS Microbiol Rev. 2018; doi:10.1093/femsre/fuy008
16. Misztal PK, Lymperopoulou DS, Adams RI, Scott RA, Lindow SE, Bruns T, et al. Emission Factors of Microbial Volatile Organic Compounds from Environmental Bacteria and Fungi. Environ Sci Technol. 2018;52: 82728282.
17. Nazir R, Warmink JA, Boersma H, van Elsas JD. Mechanisms that promote bacterial fitness in fungal-affected soil microhabitats. FEMS Microbiol Ecol. 2010;71: 169185.
18. Nazir R, Mazurier S, Yang P, Lemanceau P, van Elsas JD. The Ecological Role of Type Three Secretion Systems in the Interaction of Bacteria with Fungi in Soil and Related Habitats Is Diverse and Context-Dependent. Front Microbiol. 2017;8: 38.
19. Warmink JA, Nazir R, Corten B, van Elsas JD. Hitchhikers on the fungal highway: The helper effect for bacterial migration via fungal hyphae. Soil Biol Biochem. 2011/4;43: 760765.
20. Frey-Klett P, Burlinson P, Deveau A, Barret M, Tarkka M, Sarniguet A. Bacterial-fungal interactions: hyphens between agricultural, clinical, environmental, and food microbiologists. Microbiol Mol Biol Rev. 2011;75: 583609.
21. Garbeva P, Hordijk C, Gerards S, de Boer W. Volatile-mediated interactions between phylogenetically different soil bacteria. Front Microbiol. 2014;5: 289.
22. Kurze S, Bahl H, Dahl R, Berg G. Biological Control of Fungal Strawberry Diseases by Serratia plymuthica HRO-C48. Plant Dis. 2001;85: 529534.
23. Trunk K, Peltier J, Liu Y-C, Dill BD, Walker L, Gow NAR, et al. The type VI secretion system deploys antifungal effectors against microbial competitors. Nature Microbiology. 2018; 1.
24. Spraker JE, Wiemann P, Baccile JA, Venkatesh N, Schumacher J, Schroeder FC, et al. Conserved Responses in a War of Small Molecules between a Plant-Pathogenic Bacterium and Fungi. MBio. 2018;9. doi:10.1128/mBio.00820-18
25. Mannaa M, Oh JY, Kim KD. Biocontrol Activity of Volatile-Producing Bacillus megaterium and Pseudomonas protegens against Aspergillus flavus and Aflatoxin Production on Stored Rice Grains. Mycobiology. 2017;45: 213219.
26. Gao H, Li P, Xu X, Zeng Q, Guan W. Research on Volatile Organic Compounds From Bacillus subtilis CF-3: Biocontrol Effects on Fruit Fungal Pathogens and Dynamic Changes During Fermentation. Front Microbiol. 2018;9: 456.
27. Machado AP, Anzai M, Fischman O. Bacillus subtilis induces morphological changes in Fonsecaea pedrosoi in vitro resulting in more resistant fungal forms in vivo. J Venom Anim Toxins Incl Trop Dis. 2010;16: 592598.
28. Minerdi D, Bossi S, Gullino ML, Garibaldi A. Volatile organic compounds: a potential direct long-distance mechanism for antagonistic action of Fusarium oxysporum strain MSA 35. Environ Microbiol. 2009;11: 844854.
29. Vespermann A, Kai M, Piechulla B. Rhizobacterial volatiles affect the growth of fungi and Arabidopsis thaliana. Appl Environ Microbiol. 2007;73: 56395641.
30. Popova AA, Koksharova OA, Lipasova VA, Zaitseva JV, Katkova-Zhukotskaya OA, Eremina SI, et al. Inhibitory and Toxic Effects of Volatiles Emitted by Strains of Pseudomonas and Serratia on Growth and Survival of Selected Microorganisms, Caenorhabditis elegans, and Drosophila melanogaster. Biomed Res Int. 2014;2014. doi:10.1155/2014/125704
31. Gkarmiri K, Finlay RD, Alstrm S, Thomas E, Cubeta MA, Hgberg N. Transcriptomic changes in the plant pathogenic fungus Rhizoctonia solani AG-3 in response to the antagonistic bacteria Serratia proteamaculans and Serratia plymuthica. BMC Genomics. 2015;16: 630.
32. Woodhams DC, LaBumbard BC, Barnhart KL, Becker MH, Bletz MC, Escobar LA, et al. Prodigiosin, Violacein, and Volatile Organic Compounds Produced by Widespread Cutaneous Bacteria of Amphibians Can Inhibit Two Batrachochytrium Fungal Pathogens. Microb Ecol. 2017; doi:10.1007/s00248-017-1095-7
33. Using Serratia plymuthica to control fungal pathogens of plants. In: Loop [Internet]. [cited 25 Jun 2016]. Available: http://loop.frontiersin.org/publications/29877540
34. Schmidt R, Jager V de, Zhlke D, Wolff C, Bernhardt J, Cankar K, et al. Fungal volatile compounds induce production of the secondary metabolite Sodorifen in Serratia plymuthica PRI-2C. Sci Rep. 2017;7: 862.
35. Srobel G, Li JY, Sugawara F, Koshino H, Harper J, Hess WM. Oocydin A, a chlorinated macrocyclic lactone with potent anti-oomycete activity from Serratia marcescens. Microbiology. 1999;145 ( Pt 12): 35573564.
36. Dhar Purkayastha G, Mangar P, Saha A, Saha D. Evaluation of the biocontrol efficacy of a Serratia marcescens strain indigenous to tea rhizosphere for the management of root rot disease in tea. PLoS One. 2018;13: e0191761.
37. Hu W, Kuang F, Lu Z, Zhang N, Chen T. Killing Effects of an Isolated Serratia marcescens KH-001 on Diaphorina citri via Lowering the Endosymbiont Numbers. Front Microbiol. 2018;9: 860.
38. Duzhak AB, Panfilova ZI, Duzhak TG, Vasyunina EA, Shternshis MV. Role of prodigiosin and chitinases in antagonistic activity of the bacterium Serratia marcescens against the fungus Didymella applanata. Biochemistry . 2012;77: 910916.
39. Someya N, Nakajima M, Watanabe K, Hibi T, Akutsu K. Potential of Serratia marcescens strain B2 for biological control of rice sheath blight. Biocontrol Sci Technol. 2005;15: 105109.
40. Ipcho S, Sundelin T, Erbs G, Kistler HC, Newman M-A, Olsson S. Fungal Innate Immunity Induced by Bacterial Microbe-Associated Molecular Patterns (MAMPs). G3 . 2016;6: 15851595.
41. Soanes DM, Talbot NJ. Comparative Genome Analysis Reveals an Absence of Leucine-Rich Repeat Pattern-Recognition Receptor Proteins in the Kingdom Fungi. PLoS One. 2010;5: e12725.
42. Wheatley RE. The consequences of volatile organic compound mediated bacterial and fungal interactions. Antonie Van Leeuwenhoek. 2002;81: 357364.
43. Schmidt R, Cordovez V, de Boer W, Raaijmakers J, Garbeva P. Volatile affairs in microbial interactions. ISME J. 2015;9: 23292335.
44. Schulz-Bohm K, Martn-Snchez L, Garbeva P. Microbial Volatiles: Small Molecules with an Important Role in Intra- and Inter-Kingdom Interactions. Front Microbiol. 2017;8: 2484.
45. Ossowicki A, Jafra S, Garbeva P. The antimicrobial volatile power of the rhizospheric isolate Pseudomonas donghuensis P482. PLoS One. 2017;12: e0174362.
46. Schmidt R, Etalo DW, de Jager V, Gerards S, Zweers H, de Boer W, et al. Microbial Small Talk: Volatiles in Fungal-Bacterial Interactions. Front Microbiol. 2015;6: 1495.
47. Neupane S, Finlay RD, Alstrm S, Elfstrand M, Hgberg N. Transcriptional responses of the bacterial antagonist Serratia plymuthica to the fungal phytopathogen Rhizoctonia solani. Environ Microbiol Rep. 2015;7: 123127.
48. Kai M, Crespo E, Cristescu SM, Harren FJM, Francke W, Piechulla B. Serratia odorifera: analysis of volatile emission and biological impact of volatile compounds on Arabidopsis thaliana. Appl Microbiol Biotechnol. 2010;88: 965976.
49. Hoffman MT, Gunatilaka MK, Wijeratne K, Gunatilaka L, Arnold AE. Endohyphal bacterium enhances production of indole-3-acetic acid by a foliar fungal endophyte. PLoS One. 2013;8: e73132.
50. Taghavi T, Kim C, Rahemi A. Role of Natural Volatiles and Essential Oils in Extending Shelf Life and Controlling Postharvest Microorganisms of Small Fruits. Microorganisms. 2018;6. doi:10.3390/microorganisms6040104
51. Kanchiswamy CN, Malnoy M, Maffei ME. Chemical diversity of microbial volatiles and their potential for plant growth and productivity. Front Plant Sci. 2015;6: 151.
52. Lemfack MC, Gohlke B-O, Toguem SMT, Preissner S, Piechulla B, Preissner R. mVOC 2.0: a database of microbial volatiles. Nucleic Acids Res. 2018;46: D1261D1265.
53. Sharifi R, Ryu C-M. Sniffing bacterial volatile compounds for healthier plants. Curr Opin Plant Biol. 2018;44: 8897.
54. Minnich M. Behavior and determination of volatile organic compounds in soil: A literature review [Internet]. Lockheed Environmental Systems and Technologies Co., Las Vegas, NV (United States); 1993 May. Report No.: PB-94-100153/XAB. Available: https://www.osti.gov/biblio/5582173
55. Ashelford KE, Fry JC, Bailey MJ, Day MJ. Characterization of Serratia isolates from soil, ecological implications and transfer of Serratia proteamaculans subsp. quinovora Grimont et al. 1983 to Serratia quinivorans corrig., sp. nov. Int J Syst Evol Microbiol. 2002;52: 22812289.
56. De Vleesschauwer D, Hfte M. Using Serratia plymuthica to control fungal pathogens of plants. CAB Reviews. 2003; Available: https://biblio.ugent.be/publication/385377/file/582805
57. Gutirrez-Romn MI, Holgun-Melndez F, Bello-Mendoza R, Guilln-Navarro K, Dunn MF, Huerta-Palacios G. Production of prodigiosin and chitinases by tropical Serratia marcescens strains with potential to control plant pathogens. World J Microbiol Biotechnol. 2012;28: 145153.
58. Weise T, Thrmer A, Brady S, Kai M, Daniel R, Gottschalk G, et al. VOC emission of various Serratia species and isolates and genome analysis of Serratia plymuthica 4Rx13. FEMS Microbiol Lett. 2014;352: 4553.
59. Schulz S, Dickschat JS, Kunze B, Wagner-Dobler I, Diestel R, Sasse F. Biological activity of volatiles from marine and terrestrial bacteria. Mar Drugs. 2010;8: 29762987.
60. Herrington PR, Craig JT, Chea CY, Sheridan JE. Inhibition of spore germination by volatiles from Streptomyces griseoruber. Soil Biol Biochem. 1985;17. Available: http://agris.fao.org/agris-search/search.do?recordID=US201301443033
61. Khalid S, Baccile JA, Spraker JE, Tannous J, Imran M, Schroeder FC, et al. NRPS-Derived Isoquinolines and Lipopetides Mediate Antagonism between Plant Pathogenic Fungi and Bacteria. ACS Chem Biol. 2018;13: 171179.
62. Caballero Ortiz S, Trienens M, Pfohl K, Karlovsky P, Holighaus G, Rohlfs M. Phenotypic responses to microbial volatiles render a mold fungus more susceptible to insect damage. Ecol Evol. 2018;8: 43284339.
63. Plyuta V, Lipasova V, Popova A, Koksharova O, Kuznetsov A, Szegedi E, et al. Influence of volatile organic compounds emitted by Pseudomonas and Serratia strains on Agrobacterium tumefaciens biofilms. APMIS. 2016;124: 586594.
64. Yu K, Hamilton-Kemp TR, Archbold DD, Collins RW, Newman MC. Volatile compounds from Escherichia coli O157:H7 and their absorption by strawberry fruit. J Agric Food Chem. 2000;48: 413417.
65. Groenhagen U, Baumgartner R, Bailly A, Gardiner A, Eberl L, Schulz S, et al. Production of bioactive volatiles by different Burkholderia ambifaria strains. J Chem Ecol. 2013;39: 892906.
66. Reddy DN, Al-Rajab AJ. Chemical composition, antibacterial and antifungal activities of Ruta graveolens L. volatile oils. Cogent Chemistry. 2016;2: 1220055.
67. Giorgio A, De Stradis A, Lo Cantore P, Iacobellis NS. Biocide effects of volatile organic compounds produced by potential biocontrol rhizobacteria on Sclerotinia sclerotiorum. Front Microbiol. 2015;6: 1056.
68. Bohbot JD, Dickens JC. Insect repellents: modulators of mosquito odorant receptor activity. PLoS One. 2010;5: e12138.
69. Khan IA, Abourashed EA. Leungs Encyclopedia of Common Natural Ingredients: Used in Food, Drugs and Cosmetics. Wiley; 2009.
70. Mendes NS, Bitencourt TA, Sanches PR, Silva-Rocha R, Martinez-Rossi NM, Rossi A. Transcriptome-wide survey of gene expression changes and alternative splicing in Trichophyton rubrum in response to undecanoic acid. Sci Rep. 2018;8: 2520.
71. Aryal SK, Carter-House D, Stajich JE, Dillman AR. Microbial associates of the southern mole cricket (Scapteriscus borellii) are highly pathogenic. J Invertebr Pathol. 2017;150: 5462.
72. Na F. Identification of Two Novel Fungal Species Associated with Kuroshio Shot Hole Borer (Euwallacea sp.) and Evaluation of Novel Biological Control Method to Inhibit the Fungal Associates of the Invasive Ambrosia Beetle Species in California [Internet]. UC Riverside. 2016. Available: https://escholarship.org/uc/item/8394m1k0
73. Benny GL. Methods Used by Dr. R. K. Benjamin, and Other Mycologists, to Isolate Zygomycetes. Aliso: A Journal of Systematic and Evolutionary Botany. 2008;26: 3761.
74. Chong J, Yamamoto M, Xia J. MetaboAnalystR 2.0: From Raw Spectra to Biological Insights. Metabolites. 2019;9. doi:10.3390/metabo9030057
75. Chong J, Xia J. MetaboAnalystR: an R package for flexible and reproducible analysis of metabolomics data. Bioinformatics. 2018;34: 43134314.
76. TOXNET. In: NIH Toxicology Data Network [Internet]. 30 Aug 2006 [cited 6 Sep 2019]. Available: https://toxnet.nlm.nih.gov/cgi-bin/sis/search2/r?dbs+hsdb:@term+@rn+@rel+112-12-9


